### DISCUSSION TITLE
Being able to download local files from other Devices

### DISCUSSION THREAD

**Discussion by jWXZSCsqrpZRSVwauBjn** [Spotifyd/spotifyd#1318] [Ideas]
**Is your feature request related to a problem? Please describe.**
Support for syncing local files from other devices as explained in the [Spotify Developer docs](https://developer.spotify.com/documentation/web-api/concepts/playlists#local-files).

**Comment by eladyn** [comment]
These docs only explain, how the tracks metadata appears in API requests. AFAIK, there's no possibility to retrieve them from the device that added it.

**Comment by jWXZSCsqrpZRSVwauBjn** [comment]
it is. You add it to a playlist and make it offline available on another device. If they are in the same network, they sync local songs

**Comment by eladyn** [comment]
Oh, now it also works for me. ([reference](https://www.headphonesty.com/2022/05/spotify-local-files-not-syncing/)) I dug around a bit, but could not find any previous work to reverse engineer that protocol. And while I assume that I figured out, which TCP/UDP messages are related to syncing, I wasn't able to see how they're doing it.

<summary>Things I found out:</summary>
<details>
- When hitting the download button on the phone, it sends a UDP request to everyone on the network (port: 57621) containing `SpotUdp0` and other things.
- The desktop app responds with a similar message specifically to the requesting device (port: 57621) at least containing the number `57621` and other things.
- Then the real communication begins over TCP, where the files seem to be transmitted.

So not a lot I figured out, unfortunately.
</details>

So if anyone else is interested in reversing this protocol, this might be interesting, but for now, this is probably out of scope. (Considering that this isn't even implemented on official connect devices.)
